tests/basic.rs

```diff
@@ -77,6 +77,25 @@ fn test_round_trip_non_ascii_password() {
     )
 }
 
+#[test]
+fn test_error_on_long_password() {
+    let name = generate_random_string();
+    let entry = Entry::new(&name, &name).expect("Can't create entry");
+    let password = "VGhlcmUgb25jZSB3YXMgYSBsb25nIHBhc3N3b3JkLApzbyBsb25nIG1vc3QgdGhvdWdodCBpdCBhYnN1cmQuClRoZW4gYWxvbmUgY2FtZSBhIGhhY2tlciwKd2hvIGdhdmUgaGlzIGtleWJvYXJkIGEgY2xhY2tlciwKYW5kIHRoZSBwYXNzd29yZCdzIHJhdGUgd2FzIHByb3ZlbiB0aGlyZC4KCkxvcmVtIGlwc3VtIGRvbG9yIHNpdCBhbWV0LCBjb25zZWN0ZXR1ciBhZGlwaXNjaW5nIGVsaXQuIEZ1c2NlIGZpbmlidXMgbWFsZXN1YWRhIG1pIHNpdCBhbWV0IGZldWdpYXQuIE51bmMgdmVoaWN1bGEgb2RpbyBlZ2V0IHZlaGljdWxhIHBvc3VlcmUuIFNlZCBwdWx2aW5hciBkaWFtIHZpdGFlIHNhcGllbiBncmF2aWRhIG1hdHRpcy4gVml2YW11cyBzZWQgcHVydXMgc29sbGljaXR1ZGluLCBsYWNpbmlhIGVuaW0gZXQsIGRpZ25pc3NpbSBvcmNpLiBJbiBpZCBjdXJzdXMgc2FwaWVuLCBpbiBjb25ndWUgbWFzc2EuIFBoYXNlbGx1cyBzaXQgYW1ldCByaXN1cyBub24gc2VtIHZpdmVycmEgYXVjdG9yLiBTZWQgdGVsbHVzIG51bGxhLCB2ZXN0aWJ1bHVtIGNvbW1vZG8gY3Vyc3VzIGEsIGlhY3VsaXMgaW4gZXguClV0IGV4IGxpYmVybywgdnVscHV0YXRlIGFjIHR1cnBpcyBldSwgZWdlc3RhcyB2YXJpdXMgZHVpLiBJbnRlZ2VyIGlkIHF1YW0gZWxpdC4gUHJvaW4gZmF1Y2lidXMgZ3JhdmlkYSBtZXR1cywgcnV0cnVtIHJob25jdXMgbWV0dXMgY29uc2VxdWF0IHF1aXMuIE51bGxhIGlkIHRvcnRvciBzYXBpZW4uIFBlbGxlbnRlc3F1ZSBjb25ndWUgZXN0IGxhY3VzLCBpbiB0ZW1wb3Igb3JjaSBwb3N1ZXJlIHZlbC4gU2VkIGRhcGlidXMgb3JuYXJlIHRyaXN0aXF1ZS4gTWF1cmlzIGxlbyBudW5jLCB0aW5jaWR1bnQgZXQgc2VtIHNlZCwgbWFsZXN1YWRhIGhlbmRyZXJpdCBsb3JlbS4gUHJhZXNlbnQgZmF1Y2lidXMgcGxhY2VyYXQgdmVuZW5hdGlzLiBBbGlxdWFtIGVyYXQgdm9sdXRwYXQuIEN1cmFiaXR1ciBzYWdpdHRpcyBlbGVtZW50dW0gbGliZXJvLiBQcmFlc2VudCBzZWQgbWFsZXN1YWRhIGFyY3UsIHNlZCBsYW9yZWV0IGVyYXQuIEN1cmFiaXR1ciBldCBncmF2aWRhIHR1cnBpcy4gQWVuZWFuIGxhY2luaWEgZXggYXQgZXJvcyB2b2x1dHBhdCB1bGxhbWNvcnBlci4gQWVuZWFuIGF0IHNjZWxlcmlzcXVlIGFudGUu";
```

Comment:
This looked like base64 so I decoded it and found your limerick (which has a typo, by the way).  Cute!

Unfortunately, this is not the right way to test your improved error check.  When you encode UTF-8 as UTF-16, the number of UTF-16 bytes will never be more than twice the number of UTF-8 bytes, so if all we're going to do is test with ascii, we could have just used the old error check and divided the allowed max length by 2.  Since UTF-16 is more compact than UTF-8 for characters in the basic multilingual plane (U+0000 to U+FFFF), the right test for your improved error check is to generate a UTF-8 string that's too long in UTF-8 but whose characters are in that plane so it shrinks to being storable in UTF-16.  That is, we want a positive test as well as a negative test.

So please do the following:
- Change this test to be a positive test that puts in a too-long UTF-8 string that is storable after conversion to UTF-16.
- Update the existing validate-attributes test so that it generates an alphanumeric string that is half the max bytes + 1 byte, because when that is expanded it should be too long.

(If you're busy, you can just pull this test completely and I will write the tests.  I really appreciate you're figuring out the minimal change needed in validate_attributes.

Reply:
Thanks for the review - I've modified the tests as suggested.